mmocr/datasets/preparers/dumpers/dumpers.py

```diff
@@ -47,3 +51,122 @@ def dump(self, data: List, data_root: str, split: str):
+
+    def check_image_is_valid(self, imageBin):
+        if imageBin is None:
+            return False
+        imageBuf = np.frombuffer(imageBin, dtype=np.uint8)
+        img = cv2.imdecode(imageBuf, cv2.IMREAD_GRAYSCALE)
+        imgH, imgW = img.shape[0], img.shape[1]
+        if imgH * imgW == 0:
+            return False
+        return True
+
+    def write_cache(self, env, cache):
+        with env.begin(write=True) as txn:
+            cursor = txn.cursor()
+            cursor.putmulti(cache, dupdata=False, overwrite=True)
+
+    def parser_pack_instance(self, instance: Dict):
+        """parser an packed MMOCR format textrecog instance.
+
+        Args:
+            instance (Dict): An packed MMOCR format textrecog instance.
+                For example,
+                {
+                    "instance": [
+                        {
+                            "text": "Hello"
+                        }
+                    ],
+                    "img_path": "img1.jpg"
+                }
+        """
+        assert isinstance(instance,
+                          Dict), 'Element of data_list must be a dict'
+        assert 'img_path' in instance and 'instances' in instance, \
+            'Element of data_list must have the following keys: ' \
+            f'img_path and instances, but got {instance.keys()}'
+        assert isinstance(instance['instances'], List) and len(
+            instance['instances']) == 1
+        assert 'text' in instance['instances'][0]
+
+        img_path = instance['img_path']
+        text = instance['instances'][0]['text']
+        return img_path, text
+
+    def dump(self, data: Dict, data_root: str, split: str) -> None:
+        """Dump data to LMDB format."""
+
+        # create lmdb env
+        output_dirname = f'{self.task}_{split}.lmdb'
+        output = osp.join(data_root, output_dirname)
+        mmengine.mkdir_or_exist(output)
+        env = lmdb.open(output, map_size=self.lmdb_map_size)
+        # load data
+        if 'data_list' not in data:
+            raise ValueError('Dump data must have data_list key')
+        data_list = data['data_list']
+        cache = []
+        # index start from 1
+        cnt = 1
+        n_samples = len(data_list)
+        for d in data_list:
+            # convert both images and labels to lmdb
+            label_key = 'label-%09d'.encode(self.encoding) % cnt
+            img_name, text = self.parser_pack_instance(d)
+            img_path = osp.join(data_root, img_name)
+            if not osp.exists(img_path):
+                warnings.warn('%s does not exist' % img_path)
+                continue
+            with open(img_path, 'rb') as f:
+                image_bin = f.read()
+            if self.verify:
+                try:
+                    if not self.check_image_is_valid(image_bin):
+                        warnings.warn('%s is not a valid image' % img_path)
+                        continue
+                except Exception:
```

Comment:
Under what condition would this exception be raised? It's better not to catch the exception in order to keep the dataset full

Reply:
It's refered to https://github.com/open-mmlab/mmocr/blob/2a2cab3c8c17cac31e6d00f2b68f9eb467a15727/tools/dataset_converters/textrecog/lmdb_converter.py#L109-L115